bump gmpy2 to 2.2.1

dimpase opened the pull request:
we now have a new stable version of gmpy2

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

dimpase committed:
bump gmpy2 to 2.2.1

kwankyu reviewed (approved):
OK.

Please format the Checklist correctly.

dimpase committed:
fix/update testing of gmpy2

the main tests are now by pytest;
cython tests are done in another test dir, fixed this too.

pytest uses hypothesis, which I thus promote to standard package

dimpase committed:
spkg-configure for hypothesis

dimpase committed:
make hypothesis into a correct standard package, adjust deps

dimpase committed:
added forgotten version file

dimpase committed:
add sortedcontainers spkg

dimpase committed:
correct URL for sortedcontainers

dimpase committed:
invoke pytest at the correct directory

dimpase committed:
test_cython is only in 2.2.2, not in 2.2.1
